Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ViewTextController.java

```diff
@@ -42,7 +43,9 @@
         String respMessage = null;
         try {
             respMessage = tfManager.retrieveTextFileContent(txtId);
-            return new ResponseEntity<String>(respMessage, HttpStatus.OK);
+            String decodedRespMsg = StringEscapeUtils.unescapeXml(respMessage);
```

Comment:
Is the text still escaped when stored? It needs to be unescaped before storing, otherwise if someone submits an XML file, it won't be an actual XML file that is stored, which would be bad for reusability purposes.